src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+      case 'clear':
+        for (const prop in currentState) {
+          if (currentState.hasOwnProperty(prop)) {
+            delete currentState[prop];
+          }
+        }
+        break;
+    }
```

Comment:
You need `default` for your `switch`

Reply:
I`ll add it, but just to be sure: is it necessary to add 'default' if it doesn`t do anything?

default:
  break;

Reply:
It is a good practice - it is always better to be explicit about what your code is doing - even if you just skip the item you should do it